libs/Microsoft.MixedReality.WebRTC.Unity/Assets/Microsoft.MixedReality.WebRTC.Unity.Editor/WebcamSourceEditor.cs

```diff
@@ -0,0 +1,525 @@
+                        // without losing the value previously input. This works only while the inspector is
+                        // alive, that is while the object is select, but is better than nothing.
+                        defaultValue = newValue;
+                    }
+                    else
+                    {
+                        GUILayout.Label(_anyContent, GUILayout.Width(_anyWidth));
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// TextField with optional toggle associated with a given SerializedProperty, to enable
+        /// automatic GUI handlings like Prefab revert menu.
+        /// </summary>
+        /// <param name="stringProperty">The string property associated with the control.</param>
+        /// <param name="defaultValue">Default value if the property value null or whitespace. Assigned the new value on return if valid.</param>
+        /// <param name="label">The label to display next to the toggle control.</param>
+        private void OptionalTextField(SerializedProperty stringProperty, ref string defaultValue, GUIContent label)
+        {
+            // Note: there is a small bug in this, if the user enters a valid value on a prefab instance,
+            // validates (Return/Enter/unfocus), then select the DoubleField, then click RMB > Revert, then
+            // the property value is reset but the DoubleField value stays the same instead of being cleared.
+            // If the user does not reselect the DoubleField however, which is the most common case, things
+            // work as expected. So this rarely happens.
+
+            if (string.IsNullOrWhiteSpace(defaultValue))
+            {
+                throw new ArgumentOutOfRangeException("Default value cannot be invalid.");
+            }
+
+            using (new EditorGUILayout.HorizontalScope())
+            {
+                var rect = EditorGUILayout.GetControlRect();
+                using (new EditorGUI.PropertyScope(rect, label, stringProperty))
+                {
+                    bool hasConstraint = !string.IsNullOrWhiteSpace(stringProperty.stringValue);
+                    hasConstraint = EditorGUI.ToggleLeft(rect, label, hasConstraint);
+                    string newValue = stringProperty.stringValue;
+                    if (hasConstraint)
+                    {
+                        // Force a valid value, otherwise the edit field won't show up
+                        if (string.IsNullOrWhiteSpace(newValue))
+                        {
+                            newValue = defaultValue;
+                        }
+
+                        // Make this delayed to allow override for negative numbers before it takes effect.
+                        // This allows the user to type "-5" and only on Return/Enter or focus loss see the
+                        // value change to 0.
```

Comment:
This should be reworded.

Reply:
Something like this?
> // Make updating the value of the serialized property delayed to allow overriding the
> // value the user will input before it's assigned to the property, for validation.

Reply:
Fixed.

Reply:
I just meant that this talks about integers but it's about a text property 😄 I guess it's a bit more precise now.